test/langtools/tools/javac/processing/model/util/elements/MethodParameters/ImplicitParametersProcessor.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8292275
+ * @summary Check origin of implicit parameters
+ * @library /tools/javac/lib
+ * @modules java.compiler
+ *          jdk.compiler
+ * @build   JavacTestingAbstractProcessor ImplicitParametersProcessor
+ * @compile -processor ImplicitParametersProcessor -proc:only ClassContainer.java
+ */
```

Comment:
Good start on the annotation processor. From personal experience writing such tests and then looking at them months later, I recommend adding some kind of brief comment describing the intended semantics of the test, e.g.

"Verify enum classes and record classes have expected mandated executables."

To make the test more extensible, the expected information could be encoded in annotations; e.g. something like

@interface ExpectedInfo {
    String methodName();
    Elements.Origin origin();
}

where the record and enum were annotated in turn:

@ExpectedInfo(name="<init>", origin=MANDATED)
MyEnum {..}

etc.